src/main/java/org/springframework/cloud/deployer/scheduler/spi/kubernetes/support/RelaxedNames.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.scheduler.spi.kubernetes.support;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.util.StringUtils;
+
+/**
+ * Generates relaxed name variations from a given source.
+ *
+ * @author Phillip Webb
+ * @author Dave Syer
+ */
+public final class RelaxedNames implements Iterable<String> {
```

Comment:
i think it would be best not to copy classes over. after `RelaxedNames` was deprecated a comparable impl was made - see `org.springframework.cloud.deployer.spi.kubernetes.ImagePullPolicy.relaxedValueOf(..)`. this would also go away if we use `EntryPointStyle` and `ImagePullPolicy` as mentioned above rather than the dups